src/Common/src/System/Net/Logging/HttpEventSource.cs

```diff
@@ -0,0 +1,161 @@
+                dataDesc[1].DataPointer = (IntPtr)(arg2Ptr);
+                dataDesc[1].Size = (objName.Length + 1) * sizeof(char);
+                dataDesc[2].DataPointer = (IntPtr)(&objHash);
+                dataDesc[2].Size = sizeof(int);
+
+                WriteEventCore(URI_BASE_ADDRESS_ID, SIZEDATA, dataDesc);
+            }
+        }
+
+        [NonEvent]
+        internal static void ContentNull(object obj)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            s_log.ContentNull(LoggingHash.GetObjectName(obj), LoggingHash.HashInt(obj));
+        }
+
+        [Event(CONTENT_NULL_ID, Keywords = Keywords.Debug,
+            Level = EventLevel.Informational)]
+        internal void ContentNull(string objName, int objHash)
+        {
+            WriteEvent(CONTENT_NULL_ID, objName, objHash);
+        }
+
+        [NonEvent]
+        internal static void ClientSendCompleted(HttpClient httpClient, HttpResponseMessage response, HttpRequestMessage request)
+        {
+            if (!s_log.IsEnabled())
+            {
+                return;
+            }
+            string responseString = "";
+            if (response != null)
+            {
+                responseString = response.ToString();
+            }
+            s_log.ClientSendCompleted(LoggingHash.HashInt(request), LoggingHash.HashInt(response), responseString, LoggingHash.HashInt(httpClient));
+        }
+
+        [Event(CLIENT_SEND_COMPLETED, Keywords = Keywords.Debug,
+            Level = EventLevel.Verbose)]
+        internal unsafe void ClientSendCompleted(int httpRequestMessageHash, int httpResponseMessageHash, string responseString, int httpClientHash)
+        {
+            const int SIZEDATA = 4;
+            fixed (char* arg1Ptr = responseString)
+            {
+                EventData* dataDesc = stackalloc EventSource.EventData[SIZEDATA];
+                dataDesc[0].DataPointer = (IntPtr)(&httpRequestMessageHash);
+                dataDesc[0].Size = sizeof(int);
+                dataDesc[1].DataPointer = (IntPtr)(&httpResponseMessageHash);
+                dataDesc[1].Size = sizeof(int);
+                dataDesc[2].DataPointer = (IntPtr)(arg1Ptr);
+                dataDesc[2].Size = (responseString.Length + 1) * sizeof(char);
+                dataDesc[3].DataPointer = (IntPtr)(&httpClientHash);
+                dataDesc[3].Size = sizeof(int);
+
+                WriteEventCore(CLIENT_SEND_COMPLETED, SIZEDATA, dataDesc);
+            }
+        }
+
+        [Event(HEADERS_INVALID_VALUE_ID, Keywords = Keywords.Debug,
+            Level = EventLevel.Verbose)]
+        internal void HeadersInvalidValue(string name, string rawValue)
+        {
+            WriteEvent(HEADERS_INVALID_VALUE_ID, name, rawValue);
+        }
+
+        public class Keywords
+        {
+            public const EventKeywords Default = (EventKeywords)0x0001;
```

Comment:
GlobalLog for some of the events. For the main majority of events, I used debug since it felt that default should only be used for a few default events. But I don't have any strong argument to say a specific event should have one keyword or another. We could go through the list of events and decide if the keyword should change for some events.